maptivist/src/main/webapp/script.js

```diff
@@ -286,26 +319,133 @@ var GoogleAuth;
     }
   }
 
+
+
+
   function revokeAccess() {
     GoogleAuth.disconnect();
   }
 
   function setSigninStatus(isSignedIn) {
     var user = GoogleAuth.currentUser.get();
     var isAuthorized = user.hasGrantedScopes(SCOPE);
+    
     if (isAuthorized) {
       $('#sign-in-or-out-button').html('Sign out');
-      $('#myButton').css('display', 'block');
-      $('#auth-status').html('You are currently signed in and have granted ' +
-          'access to this app.');
+      $('#authLink').html('Sign out');
+      $('#myButton').html('Become a Maptivist');
+      $('#myButton').css('display','block');
     } else {
-      $('#sign-in-or-out-button').html('Sign In/Authorize');
-      $('#myButton').css('display', 'none');
-      $('#auth-status').html('You have not authorized this app or you are ' +
-          'signed out.');
+      $('#authLink').html('Sign in here');
+      $('#myButton').html('Sign in');
+      $('#myButton').css('display','block');
     }
   }
 
+
   function updateSigninStatus(isSignedIn) {
     setSigninStatus();
-  }
\ No newline at end of file
+  }
+
+*/
+  //Firebase
+  var firebaseConfig = {
+    apiKey: "AIzaSyDa2o12S4k-ILczNuf-WpmeY2zIFoUFxAQ",
+    authDomain: "maptivist-step-2020.firebaseapp.com",
+    databaseURL: "https://maptivist-step-2020.firebaseio.com",
+    projectId: "maptivist-step-2020",
+    storageBucket: "maptivist-step-2020.appspot.com",
+    messagingSenderId: "1032109305013",
+    appId: "1:1032109305013:web:bef8b67db0d5ae091422fc",
+    measurementId: "G-E38S0D5C9B"
+};
+
+    firebase.initializeApp(firebaseConfig);
+    //firebase.analytics();
+
+    
+var uid = firebase.auth().currentUser;
+
+
+
+function signUp() {
+    var email = document.getElementById("email");
+    var password = document.getElementById("pass");
+    const promise = firebase.auth().createUserWithEmailAndPassword(email.value, password.value);
+    promise.catch(e => alert(e.message));
+    uid = firebase.auth().currentUser;
+  }
+  
+function signIn(){
+    var email = document.getElementById("email");
+    var password = document.getElementById("pass");
+    const promise = firebase.auth().signInWithEmailAndPassword(email.value, password.value);
+    promise.catch(e => alert(e.message));
+    uid = firebase.auth().currentUser;
+
+ }
+
+function signOut(){
+    firebase.auth().signOut();
+    uid = firebase.auth().currentUser;
+    signInButtons();
+}
+
+
+
+
+function signInOrOut2() {
```

Comment:
Can you name this function something more informative?

openFormOrSignIn() perhaps?